Use the server to run womtool validation

kshakir opened the pull request.

kshakir commented:
- Added an option to disable validation due to the validation false negatives reported in #139 
- Moved tests that were calling out to the `womtool` executable from unit to integration tests
- Integration tests for validation now make calls to the cromwell server

kshakir commented on src/cromshell/utilities/cromshellconfig.py:
```diff
@@ -23,7 +23,8 @@
     "subWorkflowId",
 ]
-API_STRING = "/api/workflows/v1"
+CROMWELL_API_STRING = "/api/workflows/v1"
+WOMTOOL_API_STRING = "/api/womtool/v1"
```
One of many examples where I am open to naming suggestions. For example: here the names follow the CLI naming scheme (`cromwell` and `womtool`), but I could also see naming them after the endpoints (`workflows` and `womtool`).

lbergelson replied:
I think cromwell and womtool are very understandable.  I'm definitely used to it being called cromwell, not "workflows api"

lbergelson commented on src/cromshell/submit/command.py:
```diff
@@ -136,38 +151,54 @@ def validate_input(wdl: str, wdl_json: str, options_json: str, dependencies_zip:
         io_utils.assert_file_is_not_empty(dependencies_zip, "Dependencies Zip")
 
     # At this point, we should validate our inputs if womtool is in PATH:
```
this comment is out of date now

lbergelson reviewed (commented):
@kshakir This seems sane to me.  It nice to us the server womtool so it stays in sync. I found 1 tiny issue, and I have few questions.

1.  The reason we want to use validate before submitting is to reduce latency and get an instant result.  Does this introduce noticeable latency to the process?  

2. Is there a way to validate it WITH the dependencies? 

3.  If there is no way to validate with the dependencies zip, and it will ensure that it fails, can we make it automatically skip and notify if there is one included.

lbergelson commented on src/cromshell/submit/command.py:
```diff
@@ -136,38 +151,54 @@ def validate_input(wdl: str, wdl_json: str, options_json: str, dependencies_zip:
 
 
-def womtool_validate_wdl_and_json(wdl: str, wdl_json: str):
+def womtool_validate_wdl_and_json(
```
Lets make this -> None instead of always returning 0.

kshakir commented:
> * The reason we want to use validate before submitting is to reduce latency and get an instant result.  Does this introduce noticeable latency to the process?

I don't notice significant latency. My guess is that the time to communicate over HTTPS should be about the same latency as spinning up a JVM. So I would expect 0-to-1 second latencies for both the womtool CLI or REST API.

> * Is there a way to validate it WITH the dependencies?

Not yet as far as I can tell. I added a comment with a pointer in #139 where one could wire in the dependencies zip in the Womtool (aka Cromwell) codebase with patience.

> * If there is no way to validate with the dependencies zip, and it will ensure that it fails, can we make it automatically skip and notify if there is one included.

Added. As always feel free to refactor, edit inglish, etc.

lbergelson reviewed (approved):
@kshakir Looks good to me.
